root/programs/VB/Frameworks/Infrastructure/Business/Business/MyApsBaseLogic.vb

```diff
@@ -0,0 +1,584 @@
+
+                ' Log4Netへログ出力
+                LogIF.WarnLog("ACCESS", strLogMessage)
+            End If
+
+            ' -------------------------------------------------------------   
+        End Sub
+
+        ''' <summary>
+        ''' Ｂ層のシステム例外による異常終了の後処理を実装するUOCメソッド。
+        ''' </summary>
+        ''' <param name="parameterValue">引数クラス</param>
+        ''' <param name="returnValue">戻り値クラス</param>
+        ''' <param name="bsEx">BusinessSystemException</param>
+        ''' <remarks>業務コード親クラス１から利用される派生の末端</remarks>
+        Protected Overrides Sub UOC_ABEND(ByVal parameterValue As BaseParameterValue, ByVal returnValue As BaseReturnValue, ByVal bsEx As BusinessSystemException)
+            ' システム例外発生時の処理を実装
+            ' TODO:
+
+            ' nullチェック
+            ' なにもしない
+            If Me.perfRec Is Nothing Then
+            Else
+                ' 性能測定終了
+                Me.perfRec.EndsPerformanceRecord()
+
+                ' ACCESSログ出力-----------------------------------------------
+
+                Dim myPV As MyParameterValue = DirectCast(parameterValue, MyParameterValue)
+
+                ' ------------
+                ' メッセージ部
+                ' ------------
+                ' ユーザ名, IPアドレス, レイヤ, 
+                ' 画面名, コントロール名, メソッド名, 処理名
+                ' 処理時間（実行時間）, 処理時間（CPU時間）
+                ' エラーメッセージID, エラーメッセージ等
+                ' ------------
+                Dim strLogMessage As String = ((((("," & myPV.User.UserName & ",") & myPV.User.IPAddress & "," & "<<-----" & ",") & myPV.ScreenId & ",") & myPV.ControlId & ",") & myPV.MethodName & ",") & myPV.ActionType & "," & Convert.ToString(Me.perfRec.ExecTime) & "," & Convert.ToString(Me.perfRec.CpuTime) & "," & Convert.ToString(bsEx.messageID) & "," & Convert.ToString(bsEx.Message) & vbCr & vbLf
+                ' OriginalStackTrace（ログ出力）の品質向上
+                If Me.OriginalStackTrace = "" Then
+                    strLogMessage &= bsEx.StackTrace
+                Else
+                    strLogMessage &= Me.OriginalStackTrace
+                End If
+
+                ' Log4Netへログ出力
+                LogIF.ErrorLog("ACCESS", strLogMessage)
+            End If
+
+            ' -------------------------------------------------------------   
+        End Sub
+
+        ''' <summary>
+        ''' Ｂ層の一般的な例外による異常終了の後処理を実装するUOCメソッド。
+        ''' </summary>
+        ''' <param name="parameterValue">引数クラス</param>
+        ''' <param name="returnValue">戻り値クラス</param>
+        ''' <param name="ex">Exception</param>
+        ''' <remarks>業務コード親クラス１から利用される派生の末端</remarks>
+        Protected Overrides Sub UOC_ABEND(ByVal parameterValue As BaseParameterValue, ByRef returnValue As BaseReturnValue, ByVal ex As Exception)
+            ' 一般的な例外発生時の処理を実装
+            ' TODO:
+
+            ' nullチェック
+            If Me.perfRec Is Nothing Then
+                ' なにもしない
+
+                ' リスロー
```

Comment:
This problem has occurred by the conversion to VB from C # using the SharpDeveloper.

Currently, the base class 2 and the sample program has been converted by the SharpDeveloper. If you want to modify this entire problem, since, it is necessary to modify the conversion by SharpDeveloper. However, I believe this operation is not realistic.